### PR TITLE
Add users

### PR DESCRIPTION
**Pull request by 15lnestor**
## Description of Change
Adds the user model to the application. Also tweaks existing specs.

Right now, the only role for users is to create a league (which they will then be able to manage). Eventually, we could get to a point where a user can be a person that is in a league that doesn't manage it, but we don't need to worry about that right now.

## Alternate Designs
I haven't messed with any of the Devise configs, which I will probably need to do.

I switched the specs from using `create` to using `build` as well as cleaning up syntax a little.

## Benefits
Having `build` should speed up test (I think?) because it is not saving records into the database.

## Possible Drawbacks
N/A

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'rails_helper'
```
Comment (15lnestor) [inline spec/models/user_spec.rb]: I'm not checking any user validations - I assume Devise has some. I am going to check into exactly what it has.

**Comment by scottmascio2115** [comment]
Sounds good I can give you a run down of Devise on monday. In the mean time check out database cleaner. [here](http://tomdallimore.com/blog/taking-the-test-trash-out-with-databasecleaner-and-rspec/) should be a good addition to your testing suite

**Comment by 15lnestor** [comment]
Ok cool. This weekend I'm going to start working on controllers for everything. Maybe views too.

**Comment by scottmascio2115** [comment]
Rebuilding rails http://rebuilding-rails.com/

**Comment by scottmascio2115** [comment]
@15lnestor Following up with the conversation of what day to day coding life looks like I would suggest [this article](https://simpleprogrammer.com/2017/02/27/what-software-developers-do-at-work/). I would also get away from the notion that writing a lot of code in a day or week is more productive than not writing code. Taking that extra hour to explore and architect the proper solution rather then just doing the first thing that comes to your head will save tons of time in the future with technical debt and potential bugs. Lastly who wants to be a code monkey "A programmer who isn't actually involved in any aspect of conceptual or design work, but simply writes code to specifications given." Ive been a code monkey at jobs before where you just crank through features. Those jobs suck IMO. If you really want to grow as a software developer take the extra time to figure out if what you are doing isnt just the quickest easiest way to do it but also the right way

**Comment by 15lnestor** [comment]
Thanks for those two links. Is there anything I should do with this PR?

**Comment by scottmascio2115** [comment]
lgtm

**Comment by scottmascio2115** [comment]
nope merge it in

**Pull request merged**